### DISCUSSION TITLE
Ogólna dokumentacja do Frontendu

### DISCUSSION THREAD

**Discussion by mattyl006** [emkarcinos/WMIAdventure#415] [Documentation]
Zróbcie mi review tego taska z dokumentacją do Frontendu. I dajcie znać co tam dopisać jeszcze.

**Comment by mattyl006** [comment]
Pewnie np. tego dockera do instrukcji. Jak korzystacie z dockera do frontendu, to poprostu wystarczy przed npm start - `docker-compose up -d`? Aby to .env działało i w ogóle.

**Reply by emkarcinos** [reply]
Nie, te wszystkie komendy uruchamia docker. Żadnego npm nie używasz. Zobacz sobie jak jest na backendzie, 
![image](https://user-images.githubusercontent.com/11946540/126775128-4dc8103e-1125-4e48-9b68-c0793bc98f37.png)

Jedynie czym uruchamianie frontendu będzie się różnić to tymi argumentami (we frontendzie jest tylko API_URL)

**Reply by mattyl006** [reply]
Zaaktualizowałem

**Reply by mattyl006** [reply]
Ale nie działa mi localhost, po tym run docker

**Reply by emkarcinos** [reply]
hmm, u mnie wszystko działa, a co Ci się za komunikat błędu wyświetla?

**Reply by mattyl006** [reply]
wyświetla się że jest ok, ale na localhost jest 404, myślę że to może być jakiś problem u mnie na Manjaro z permissions do portu 80

**Comment by emkarcinos** [comment]
Wszystko wygląda okej, tylko mało tego jest w porównaniu tego ile mamy już kodu. Nie wiem na ile chcesz to aktualizować na bieżąco, ale dobrze byłoby, gdyby każdy komponent był chociaż jakoś krótko opisany. Po to się je robi, żeby z nich korzystać w przyszłości, a przyszli programiści teraz nie będą nawet wiedzieli jakie komponenty w ogóle istnieją

**Reply by mattyl006** [reply]
Task był do ogólnej dokumentacji

**Reply by mattyl006** [reply]
Na dole w tych *pages* będą wszystkie komponenty

**Reply by emkarcinos** [reply]
jasne, no to w takim razie wszystko jest git

**Comment by itchystone** [comment]
Dla mnie spoko, tylko podałbym od razu konkretnie w jakim katalogu wywoływać te komendy, żeby nie trzeba było szukać:
![image](https://user-images.githubusercontent.com/43492764/126862909-e5eca2cd-f443-4d6e-8fa3-8032490a10f4.png)

**Reply by mattyl006** [reply]
Tak będzie ok? 
![image](https://user-images.githubusercontent.com/51495505/126867932-6772e64e-1b3f-4e88-a01e-b8b0175dae86.png)

**Reply by itchystone** [reply]
git

**Reply by emkarcinos** [reply]
A, zapomniałem jeszcze dopisać, że jak odpalamy dockera komendą `docker run` (bez compose) to musimy podać jeszcze `-p 127.0.0.1:80:80` żeby otworzyć port dla przeglądarki, bo inaczej nie będzie się dało połączyc.

**Comment by mattyl006** [comment]
Dodałem jeszcze opcje bez dockera, może ona zostać?

**Reply by itchystone** [reply]
Chyba powinna zostać, niektórzy mogą nie mieć dockera
